src/ui/public/share/directives/share_object_url.js

```diff
@@ -71,12 +74,15 @@ app.directive('shareObjectUrl', function (Private, Notifier) {
       };
 
       // since getUrl() is called within a watcher we cache the unhashing step
-      const unhashStatesInAbsUrl = memoize((absUrl) => {
-        return Private(UnhashStatesProvider).inAbsUrl(absUrl);
+      const unhashAndCacheUrl = memoize((urlWithHashes, unhashableStates) => {
```

Comment:
`memoize` only caches based on the first argument (and should really throw when more than one is passed) so we should probably move the `getUnhashableStates()` call inside.